ui/admin/app/components/form/worker/create-worker-led/index.hbs

```diff
@@ -45,13 +45,72 @@
       />
     {{/if}}
 
-    <form.input
-      @name='worker_tags'
-      @value={{this.workerTags}}
-      @label={{t 'resources.worker.form.worker_tags.label'}}
-      @helperText={{t 'resources.worker.form.worker_tags.help'}}
-      placeholder='dev-worker, ubuntu'
-    />
+    <Rose::Form::Fieldset>
+      <:title>{{t 'resources.worker.form.worker_tags.label'}}</:title>
+      <:description>{{t
+          'resources.worker.form.worker_tags.help'
+        }}</:description>
+      <:body>
+        <Rose::Table @style='condensed' as |table|>
+          <table.body as |body|>
+            {{#each this.workerTags as |tags index|}}
+              <body.row as |row|>
+                <row.cell>
+                  <form.input
+                    @type='text'
+                    @value={{get tags 0}}
+                    {{on 'input' (fn this.updateWorkerTag index 0)}}
```

Comment:
If the tag is an object with tracked properties, manually handling the event might be unnecessary.

Reply:
> Wow great working getting this together so quick. What do you think about representing key/value pairs as and array of objects rather than arrays? That might make the code read a little nicer than using indices. e.g. `@value={{tags.key}}`

I had originally tried using an object (for the entire structure, in place of an array of arrays) but handling tracking/binding of the keys was a massive pain and I couldn't figure out a good solution to doing it nicely.

When you say an object, do you mean making a new class that represents the tag (basically a tuple)? If so, I definitely could do that but where should the file live? I originally was thinking of doing something like that but wasn't sure what the convention was.

Reply:
Interesting stuff!! I will subscribe to this thread to follow up what solution finally implement, since this specific topic is very interesting 😄

Reply:
I made a tag class inside the component class file (temporarily, not sure where else to put it, though it is quite specific to this use case).

It definitely does make it easier to read and is cleaner as I don't have to manually handle the tracking now.

Let me know what you think.